lib/ruby_lsp/tapioca/addon.rb

```diff
@@ -106,6 +111,60 @@ def workspace_did_change_watched_files(changes)
           constants: constants,
         )
       end
+
+      private
+
+      sig { void }
+      def handle_gemfile_changes
+        return unless File.exist?(".git") && File.exist?(".ruby-lsp/shutdown-timestamp")
+
+        git_timestamp = fetch_last_git_operation_time
+        return unless git_timestamp
+
+        ruby_lsp_stop_time = Time.iso8601(File.read(".ruby-lsp/shutdown-timestamp"))
+
+        $stderr.puts("ruby_lsp_stop_time: #{ruby_lsp_stop_time}") # TODO: Remove
+        $stderr.puts("git_timestamp: #{git_timestamp}") # TODO: Remove
+
+        # If the Ruby LSP was stopped shortly after the last git checkout/pull operation, we don't need to regenerate
+        # RBIs since any Gemfile.lock changes were likely from version control, not from running bundle install
+        return if (ruby_lsp_stop_time - git_timestamp) <= GIT_OPERATION_THRESHOLD
+
+        process_gemfile_changes
+      end
+
+      sig { returns(T.nilable(Time)) }
+      def fetch_last_git_operation_time
+        git_reflog_output = %x(git reflog --date=iso | grep -E "checkout|pull" | head -n 1).strip
```

Comment:
```suggestion
        git_reflog_output = %x(git reflog --date=iso --grep-reflog='^\(checkout\|pull\)' -1).strip
```
This approach has a couple of advantages:

* It helps prevent mismatches due to a commit messages co-incidentally containing the word 'checkout' or 'pull'. (The `^` means match only at the start of the line).
* It avoids the need for external programs which may not be available on Windows.